### ISSUE TITLE
Add except: ["after-rule"] to rule-nested-empty-line-before

### ISSUE THREAD

**Issue by rontav** [stylelint/stylelint#2179]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A feature request.

> Which rule, if any, is this issue related to?

`rule-nested-empty-line-before`

> What CSS is needed to reproduce this issue?

```css
a {
  color: pink;
            /* <----- Force ONLY this empty line  */
  b {
    color: red;
  }        /* No new line is required between nested selectors */
  c {
    color: blue;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

This should do what I describe:

```json
{
  "rules": {
    "rule-nested-empty-line-before": [
      "never-multi-line",
      {
        "except": ["first-selector"]
      }
    ]
  }
}
```

> Which version of stylelint are you using?

`7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Visual Studio Code plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS and nesting.

> What did you expect to happen?

To be able to set an except for this case.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
1:21    Unexpected empty line before nested rule (rule-nested-empty-line-before)
```

**Comment by jeddy3** [comment]
@rontav Thank you for the suggestion and for using the template.

I think adding an `except: ["after-rule"]` option would be consistent with our other options and I believe would satisfy your request e.g.

```json
{
  "rules": {
    "rule-nested-empty-line-before": [
      "always-multi-line",
      {
        "except": ["after-rule"]
      }
    ]
  }
}
```

Would be valid for:

```css
a {
  color: pink;
            /* <----- Force ONLY this empty line  */
  b {
    color: red;
  }        /* No new line is required between nested selectors */
  c {
    color: blue;
  }
}
```

And for other non-rule things like custom properties e.g.:

```css
a {
  --custom-prop: pink;
            /* <----- Force ONLY this empty line  */
  b {
    color: red;
  }        /* No new line is required between nested selectors */
  c {
    color: blue;
  }
}
```

**Comment by jeddy3** [comment]
@rontav As no objections were raised feel free to contribute this option. There are instructions in the developer guide for [adding the option](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-a-option-to-an-existing-rule) and for [running tests](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#running-tests).

**Comment by jeddy3** [comment]
Done in #2188
